### PR TITLE
Using go:embed

### PR DESCRIPTION
**Pull request by victortoso**
**What this PR does / why we need it**:

This PR is to propose using [embed](https://pkg.go.dev/embed) to move files from strings to actual files. The contents are loaded at compile time, so we shouldn't have lower performance. It actually can improve performance in some places that we load the files are runtime, like in [capabilities_test.go](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/node-labeller/capabilities_test.go#L137).

So, I'm starting the proposal with some changes in virtwrap by using `testdata` directory so we always have `go:embed testdata/path-to-file`.

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/converter/BUILD.bazel [hunk 1]
```diff
@@ -47,6 +47,12 @@ go_test(
         "converter_test.go",
     ],
     embed = [":go_default_library"],
+    embedsrcs = [
+        "embed/template_domain_ppc64le.xml.tmpl",
+        "embed/template_domain_x86_64.xml.tmpl",
+        "embed/template_domain_arm64.xml.tmpl",
+        "embed/template_domain_x86_64_root.xml.tmpl",
```
Comment (vasiliy-ul) [inline pkg/virt-launcher/virtwrap/converter/BUILD.bazel:54]: nit: I would omit `template_` prefix. IMHO the intension is clear from the `.tmpl` extensions,
```suggestion
        "embed/domain_ppc64le.xml.tmpl",
        "embed/domain_x86_64.xml.tmpl",
        "embed/domain_arm64.xml.tmpl",
        "embed/domain_x86_64_root.xml.tmpl",
```

**Review by vasiliy-ul — COMMENTED** [review]
A heavy +1 from my side for the idea. Having those huge domain-xml-strings in the sources makes it difficult to navigate the code. With `embed` it is much better IMHO.

I only have a concern that ideally it would be nice to have a consistency in the project. E.g. now we already use the approach with `testdata` in some other places to store xmls or other stuff for the tests:
```
$ find . -name "*testdata*"
./pkg/monitoring/domainstats/downwardmetrics/testdata
./pkg/virt-handler/isolation/testdata
./pkg/virt-handler/node-labeller/testdata
./pkg/virt-handler/heartbeat/testdata
./pkg/downwardmetrics/vhostmd/testdata
```
In general, I am fine with using `embed`. Shall we adopt it across all the tests (not proposing doing it within this PR, ofc)?

File: pkg/virt-launcher/virtwrap/converter/BUILD.bazel (hunk 1 above)
Comment (victortoso) [reply pkg/virt-launcher/virtwrap/converter/BUILD.bazel:54]: Agreed. I'll change it.

**Comment by victortoso** [comment]
> I only have a concern that ideally it would be nice to have a consistency in the project. E.g. now we already use the approach with testdata in some other places to store xmls or other stuff for the tests:

100% agree with consistency. I did not know we had already a directory like that. Let's use it! So, I'll change it to `testdata`

File: pkg/virt-launcher/virtwrap/BUILD.bazel [hunk 2]
```diff
@@ -71,7 +71,9 @@ go_test(
         "nichotplug_test.go",
         "virtwrap_suite_test.go",
     ],
+    data = glob(["testdata/**"]),
     embed = [":go_default_library"],
+    embedsrcs = ["testdata/migration_domain.xml"],
```
Comment (vasiliy-ul) [inline pkg/virt-launcher/virtwrap/BUILD.bazel:76]: I don't think you need to have both ` data = glob(["testdata/**"])` and `embedsrcs = ["testdata/migration_domain.xml"]`. AFAIK, `data = ...` is needed only if you want to reference the file in runtime:

https://github.com/kubevirt/kubevirt/blob/d5485fd59023c8bff60b1c7f9cfb55a1c061906c/pkg/virt-handler/node-labeller/capabilities_test.go#L16

IMHO, `embedsrc` should be enough.

File: pkg/virt-launcher/virtwrap/api/BUILD.bazel [hunk 3]
```diff
@@ -31,7 +31,13 @@ go_test(
         "defaults_test.go",
         "schema_test.go",
     ],
+    data = glob(["testdata/**"]),
```
Comment (vasiliy-ul) [inline pkg/virt-launcher/virtwrap/api/BUILD.bazel:34]: Same here: probably not needed.

File: pkg/virt-launcher/virtwrap/BUILD.bazel (hunk 2 above)
Comment (victortoso) [reply pkg/virt-launcher/virtwrap/BUILD.bazel:76]: I did not edit those myself, they were generated by `make bazel-generate` on each per commit.
Testing here, removing it by hand and running `bazel-generate` reintroduces it again.

It is also the same in the other places that uses `testdata`, see:

```bash
toso@tapioca ~/s/k/kubevirt (use-embed)> grep -rniI "glob.*testdata" *
pkg/monitoring/domainstats/downwardmetrics/BUILD.bazel:33:    data = glob(["testdata/**"]),
pkg/virt-handler/isolation/BUILD.bazel:41:    data = glob(["testdata/**"]),
pkg/virt-handler/heartbeat/BUILD.bazel:27:    data = glob(["testdata/**"]),
pkg/virt-handler/node-labeller/BUILD.bazel:42:    data = glob(["testdata/**"]),
pkg/virt-launcher/virtwrap/api/BUILD.bazel:34:    data = glob(["testdata/**"]),
pkg/virt-launcher/virtwrap/converter/BUILD.bazel:49:    data = glob(["testdata/**"]),
pkg/virt-launcher/virtwrap/launchsecurity/BUILD.bazel:17:    data = glob(["testdata/**"]),
pkg/virt-launcher/virtwrap/BUILD.bazel:74:    data = glob(["testdata/**"]),
pkg/downwardmetrics/vhostmd/BUILD.bazel:23:    data = glob(["testdata/**"]),
```

File: pkg/virt-launcher/virtwrap/api/BUILD.bazel (hunk 3 above)
Comment (victortoso) [reply pkg/virt-launcher/virtwrap/api/BUILD.bazel:34]: I'm seeing that `testdata`  is not random name, but actually treated specially. Seems Go [ignores](https://pkg.go.dev/cmd/go#hdr-Test_packages) testdata directories.

> The go tool will ignore a directory named "testdata", making it available to hold ancillary data needed by the tests. 

That's likely why `testdata` generates different bazel build then when I had `embed` directory.

File: pkg/virt-launcher/virtwrap/BUILD.bazel (hunk 2 above)
Comment (victortoso) [reply pkg/virt-launcher/virtwrap/BUILD.bazel:76]: So, if `data = ...` is not needed, we should tweak bazel generation and likely apply everywhere. I"d vote for keeping it now for consistency.

**Comment by victortoso** [comment]
/retest-required

Thanks Vasiliy, for the reviews!

File: pkg/virt-launcher/virtwrap/BUILD.bazel (hunk 2 above)
Comment (vasiliy-ul) [reply pkg/virt-launcher/virtwrap/BUILD.bazel:76]: Okay, I see. I was not aware that it was generated by bazel. Let’s keep it then. Thanks.

**Review by vasiliy-ul — COMMENTED** [review]
/approve

**Review by rmohr — COMMENTED** [review]
/lgtm

**Pull request merged**